docs/TopK_Commands.md

```diff
@@ -0,0 +1,196 @@
+# RedisBloom TopK Filter Command Documentation
+
+Based on paper - **HeavyKeeper: An Accurate Algorithm for Finding Top-k Elephant Flows.**
+
+Paper and additional information can be found [here](https://www.usenix.org/conference/atc18/presentation/gong).
+
+***
+
+## TOPK.RESERVE
+
+Initializes a TopK with specified parameters.
+
+```sql
+BF.RESERVE key topk width depth decay
+```
+
+### Parameters
+
+* **key**: Key under which the sketch is to be found.
+* **topk**: Number of top occurring items to keep.
+* **width**: Number of counters kept in each array.
+* **Depth**: Number of arrays.
+* **Decay**: The probability of reducing a counter in an occupied bucket. It is raised to power of it's counter (decay ^ bucket[i].counter). Therefore, as the counter gets higher, the chance of a reduction is being reduced.
+
+### Complexity
+
+O(1)
+
+### Return
+
+OK on success, error otherwise
+
+#### Reserve example
+
+```sql
+TOPK.RESERVE test 50 2000 7 0.925
+```
+
+***
+
+## TOPK.ADD
+
+Adds an item to the data structure.
+
+```sql
+TOPK.ADD key item [item ...]
+```
+
+### Parameters
+
+* **key**: Name of sketch where item is added.
+* **item**: Item to be added.
+
+### Complexity
+
+O(k + depth)
+
+### Return
+
+OK on success
+
+#### Reserve example
+
+```sql
+TOPK.ADD test foo bar 42
+```
+
+***
+
+## TOPK.QUERY
+
+Checks whether an item is one of Top-K items.
```

Comment:
I think the documentation should be more clear about which commands accept N arguments and return N-length arrays.

Reply:
How is it now?

Reply:
Much better, thanks!